swarm/src/connection/pool.rs

```diff
@@ -517,12 +516,14 @@ where
             waker.wake();
         }
 
-        let connection = Connection::new(
+        let span = tracing::error_span!("connection", id = %id, peer = %obtained_peer_id, remote_address = %endpoint.get_remote_address());
+        let connection: Connection<THandler> = Connection::new(
```

Comment:
Shouldn't need the type-hint?

```suggestion
        let connection = Connection::new(
```